commons/src/main/java/com/inrupt/rdf/wrapping/commons/ObjectSet.java

```diff
@@ -133,12 +135,16 @@ public boolean contains(final Object o) {
 
     @Override
     public Iterator<T> iterator() {
-        return values().iterator();
+        try (final Stream<T> stream = values()) {
+            return stream.iterator();
```

Comment:
What happens to the iterator when the stream is closed?

I don't imagine that the iterator consumes the entire stream before starting.
The iterator might fail at some later point in time, when the underlying stream has already closed.

Reply:
Existing test coverage is pretty naive, i.e. always only one item.
https://github.com/inrupt/rdf-wrapping-java/blob/4f73372d63a0a6c50cada0b2669109a9268b56ad/test/base/src/main/java/com/inrupt/rdf/wrapping/test/base/ObjectSetBase.java#L243-L277

Reply:
tl;dr @langsamu is correct -- the iterator will fail after the stream is closed.

I created a local test with an iterator from a stream that is closed in a similar manner. Then, when consuming the iterator, I get this exception:

    java.lang.IllegalStateException: source already consumed or closed

I will re-work this. The challenge is still that RDF4J requires that streams get closed. I think the only solution is to force it into an in-memory construct where the resource is cleaned up automatically. I will also expand the test coverage.

Reply:
I ended up with an approach that looks like this:

```java
try (Stream<T> stream = values()) {
    return stream.collect(Collectors.toList()).stream().iterator();
}
```

I don't especially love it, but it handles the case where the stream comes from an underlying closeable resource. The extra `.stream()` segment is there to avoid changing the exception expectation (keeping UnsupportedOperationException instead of changing to IllegalStateOperation).

Reply:
With https://github.com/inrupt/rdf-wrapping-java/pull/156/commits/73ce0e411193716ce50563bdf111d6786508cf25 we avoid the extra `.stream()` call, which I am much happier with